### PR TITLE
Improve by memo

### PR DESCRIPTION
**Pull request by FredPeal**

### CODE REVIEW

File: src/Models/Messages.php [hunk 1]
```diff
@@ -355,6 +356,7 @@ public function afterSave()
         if ($this->hasParent()) {
             ElasticMessages::dispatch($this->getParentMessage());
         }
+        Interactions::add($this->users, $this, 'save-message');
```
Comment (kaioken) [inline src/Models/Messages.php]: @FredPeal use constant

File: src/Models/UsersReactions.php [hunk 2]
```diff
@@ -38,4 +49,16 @@ public function initialize()
...
+
+
+    /**
+     * afterSave.
+     *
+     * @return void
+     */
+    public function afterSave()
+    {
+        $users = Users::findFirstOrFail($this->users_id);
```
Comment (kaioken) [inline src/Models/UsersReactions.php:64]: @FredPeal why not use the relationship?

File: src/Models/UsersReactions.php [hunk 3]
```diff
@@ -38,4 +49,16 @@ public function initialize()
...
+        Interactions::add($users, $this->entity_namespace::findFirst($this->entity_id), UsersInteractions::REACTION);
```
Comment (kaioken) [inline src/Models/UsersReactions.php]: @FredPeal encapsulate this metho, you are using it in multiple  location
$this->entity_namespace::findFirst($this->entity_id)

**Review by kaioken — CHANGES_REQUESTED** [review]

File: src/Models/UsersReactions.php (hunk 2 above)
Comment (FredPeal) [reply src/Models/UsersReactions.php:64]: Because the relationships are load in the afterFetch method

File: src/Traits/EntityData.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+trait EntityData
+{
+    /**
+     * retrieveEntityData.
+     *
+     * @return ModelInterface
+     */
+    public function retrieveEntityData() : ModelInterface
+    {
+        return $this->entity_namespace::findFirst($this->entity_id);
```
Comment (kaioken) [inline src/Traits/EntityData.php:18]: @FredPeal this can return null

**Review by kaioken — CHANGES_REQUESTED** [review]

**Review by kaioken — APPROVED** [review]

**Pull request merged**